### PR TITLE
Added LSM and Updated CGLS to work with broadcasted arrays

### PR DESCRIPTION
**Pull request by rohanbabbar04**
For #46
- Updated CGLS to work with broadcasted arrays
- Added a test to verify the above using MPIHStack
- Added LSM in tutorials section.

### CODE REVIEW

File: tutorials/plot_lsm.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    axs[0].axis("tight")
+    axs[1].imshow(d_adj[0, :, :300].T, cmap="gray", vmin=-d_adj.max(), vmax=d_adj.max())
+    axs[1].set_title(r"$d_{adj}$")
+    axs[1].axis("tight")
+    axs[2].imshow(d_inv[0, :, :300].T, cmap="gray", vmin=-d.max(), vmax=d.max())
+    axs[2].set_title(r"$d_{inv}$")
+    axs[2].axis("tight")
+
+    fig3, axs = plt.subplots(1, 3, figsize=(10, 3))
+    axs[0].imshow(d[ns // 2, :, :300].T, cmap="gray", vmin=-d.max(), vmax=d.max())
```
Comment (mrava87) [inline tutorials/plot_lsm.py:199]: nstot?

File: tests/test_solver.py [hunk 2]
```diff
@@ -108,3 +108,46 @@ def test_cgls(par):
         y1 = BDiag * x_global
         xinv1 = pylops.cgls(BDiag, y1, x0=x0, niter=par["nx"], tol=1e-5, show=True)[0]
         assert_allclose(xinv_array, xinv1, rtol=1e-14)
+
+
+@pytest.mark.mpi(min_size=2)
+@pytest.mark.parametrize(
+    "par", [(par1), (par1j), (par2), (par2j), (par3), (par3j), (par4), (par4j)]
+)
+def test_cgls_broadcast(par):
```
[the remark this answers is not in the document]
Comment (mrava87) [reply tests/test_solver.py:117]: Also, it may be good to call this test `test_cgls_broadcastdata`

File: tutorials/plot_lsm.py (hunk 1 above)
Comment (rohanbabbar04) [reply tutorials/plot_lsm.py]: Missed this one, thank for pointing this out.
Corrected it...

File: tests/test_solver.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply tests/test_solver.py]: Done...
Also added the example for `broadcastmodel`

**Comment by mrava87** [comment]
@rohanbabbar04 I noticed one more thing in the test_solver, I'll raise an issue and we can discuss there :)

**Comment by rohanbabbar04** [comment]
> @rohanbabbar04 I noticed one more thing in the test_solver, I'll raise an issue and we can discuss there :)

Sure...

**Pull request merged**